File: hw1/MyApplication3m1d/app/src/main/java/com/example/myapplication3m1d/MainActivity.java
package com.example.myapplication3m1d;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Parcelable;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    ListView listView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listView=findViewById(R.id.listview);
        ArrayList<Summary> datas=new ArrayList<>();
        datas.add(new Summary("简单","1.两数之和","题解16,152","通过率52.3%","1.给定一个整数数组nums 和一个整数目标值target ，请你在该数组中找出和为目标值target的那两个整数，并返回它们的数组下标。\n\n" +
                "你可以假设每种输入只会对应一个答案。但是，数组中同一个元素在答案里不能重复出现。\n\n" +
                "你可以按任意顺序返回答案。\n"));
        datas.add(new Summary("中等","1.两数相加","题解6,152","48.4%","2.给定一个整数数组nums 和一个整数目标值target ，请你在该数组中找出和为目标值target的那两个整数，并返回它们的数组下标。\n\n" +
                "你可以按任意顺序返回答案。\n"));
        datas.add(new Summary("困难","1.两数乘积","题解12,752","通过率32.7%","3.给定一个整数数组nums 和一个整数目标值target ，请你在该数组中找出和为目标值target的那两个整数，并返回它们的数组下标。\n\n" +
                "你可以假设每种输入只会对应一个答案。但是，数组中同一个元素在答案里不能重复出现。\n\n" +
                "你可以按任意顺序返回答案。\n"));
        datas.add(new Summary("简单","1.两数比较","题解18,142","通过率56.6%","4.给定一个整数数组nums 和一个整数目标值target ，请你在该数组中找出和为目标值target的那两个整数，并返回它们的数组下标。\n\n" +
                "你可以假设每种输入只会对应一个答案。但是，数组中同一个元素在答案里不能重复出现。\n\n" +
                "你可以按任意顺序返回答案。\n"));
        MAdapter adapter=new MAdapter(getApplicationContext(),0,datas);
        listView.setAdapter(adapter);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Intent intent=new Intent(MainActivity.this,DetailActivity.class);
                intent.putExtra("data", datas.get(i));
                startActivity(intent);
            }
        });
    }
}

File: android_work/StudentInfoManager/app/src/main/java/com/iuw/app/MainActivity.kt
package com.iuw.app

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.iuw.app.adapter.StudentAdapter
import com.iuw.app.db.DBUtil
import com.iuw.app.db.bean.Student
import com.iuw.app.event.ChangeEvent
import kotlinx.android.synthetic.main.activity_main.*
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class MainActivity : AppCompatActivity(), Toolbar.OnMenuItemClickListener {
    private val studentDate = ArrayList<Student>()
    private val adapter = StudentAdapter(studentDate)

    /**
     * activity生命周期
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //设置页面内容
        setContentView(R.layout.activity_main)
        //注册 用于接受页面更新事件
        EventBus.getDefault().register(this)
        //查询数据库中的所有学生信息
        val allStudent = DBUtil.studentDao().loadAllStudent()
        studentDate.addAll(allStudent)
        studentRecyclerView.adapter = adapter
        //添加菜单到toolBar中
        toolBar.inflateMenu(R.menu.menu_student )
        //为toolbar设置item的点击事件
        toolBar.setOnMenuItemClickListener(this)

    }

    override fun onMenuItemClick(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.add -> {
                if (DBUtil.lessonDao().loadAll().isEmpty()) {
                    Toast.makeText(this, "请先添加教师信息", Toast.LENGTH_SHORT).show()
                    return false
                }
                EditInfoActivity.toMeInsert(this, EditInfoActivity.STUDENT)
            }
            R.id.viewTeachers -> {
                startActivity(Intent(this, TeacherListActivity::class.java))
            }
            else -> {
                EditInfoActivity.toMeInsert(this, EditInfoActivity.TEACHER)
            }
        }
        return false
    }

    //订阅发布的内容(当EventBus.getDefault().post(ChangeEvent())被调用时候，
    // 加了Subscribe注解并且参数类型为ChangeEvent的方法就会被调用)
    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onChangeEvent(event: ChangeEvent) {
        studentDate.clear()
        val allStudent = DBUtil.studentDao().loadAllStudent()
        studentDate.addAll(allStudent)
        adapter.notifyDataSetChanged()
    }

}